componenets/org.wso2.carbon.identity.application.authenticator.naver/src/main/java/org.wso2.carbon.identity.application.authenticator.naver/NaverCustomAuthenticator.java

```diff
@@ -46,6 +87,69 @@ public class NaverCustomAuthenticator extends Oauth2GenericAuthenticator {
 
     private static final long serialVersionUID = 6614257960044886319L;
 
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        return isNativeSDKBasedFederationCall(request) || super.canHandle(request);
+    }
+
+    /**
+     * Check whether the authentication is based on API.
+     *
+     * @return true since API based authentication is supported.
+     */
+    @Override
+    public boolean isAPIBasedAuthenticationSupported() {
+
+        return true;
+    }
+
+    /**
+     * Get the i18n key defined to represent the authenticator name.
+     *
+     * @return the 118n key.
+     */
+    @Override
+    public String getI18nKey() {
+
+        return NaverCustomAuthenticatorConstants.AUTHENTICATOR_I18N_KEY;
+    }
+
+    @Override
+    protected void initiateAuthenticationRequest(HttpServletRequest request, HttpServletResponse response, AuthenticationContext context) throws AuthenticationFailedException {
+
+        if (logger.isDebugEnabled()) {
```

Comment:
Since this is a simple debug log without any String concatenations, shall we remove the if condition?

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-naver/pull/13/commits/0bd7d3f95c3c1490a22874034a254d73712cecae